src/WebApi/Controllers/UsersController.cs

```diff
@@ -210,6 +210,38 @@ public Task<ActionResult<Result<CharacterCharacteristicsViewModel>>> GetCharacte
         }));
     }
 
+    /// <summary>
+    /// Get character competitive rating.
+    /// </summary>
+    /// <param name="id">Character id.</param>
+    /// <returns>The character rating.</returns>
+    /// <response code="200">Ok.</response>
+    [HttpGet("self/characters/{id}/competitiverating")]
+    public Task<ActionResult<Result<CharacterCompetitiveRatingViewModel>>> GetCharacterRating([FromRoute] int id)
+    {
+        return ResultToActionAsync(Mediator.Send(new GetUserCharacterCompetitiveRatingQuery
+        {
+            UserId = CurrentUser.User!.Id,
+            CharacterId = id,
+        }));
+    }
+
+    /// <summary>
+    /// Get top character competitive ratings.
+    /// </summary>
+    /// <returns>The top character competitive ratings.</returns>
+    /// <response code="200">Ok.</response>
+    [HttpGet("characters/topcompetitiverating")]
```

Comment:
I believe that you need a separate controller for this call, since it is in no way related to the `user`. 
ex: `Ladder`, `Rating`, idk

Reply:
done